mne/io/eyetrack/eyetrack.py

```diff
@@ -0,0 +1,243 @@
+                                             '%a %b %d %H:%M:%S %Y')
+            meas_date = meas_date.replace(tzinfo=dt.timezone.utc)
+
+        if meas_date is None:
+            warn("Extraction of measurement date from asc file failed. "
+                 "Please report this as a github issue. "
+                 "The date is being set to January 1st, 2000, ")
+            meas_date = dt.datetime(2000, 1, 1, 0, 0, 0,
+                                    tzinfo=dt.timezone.utc)
+
+        # set parameter
+        ch_names = []
+        if pos:
+            ch_names.append('X')
+            ch_names.append('Y')
+        if pupil:
+            ch_names.append('Pupil')
+        if eye == 'LEFT':
+            ch_names = ['L' + ch for ch in ch_names]
+        elif eye == 'RIGHT':
+            ch_names = ['R' + ch for ch in ch_names]
+        elif eye == 'BINO':
+            ch_names = [['R' + ch, 'L' + ch] for ch in ch_names]
+            ch_names = [x for xs in ch_names for x in xs]
+        ch_names.sort()
+
+        n_chan = len(ch_names)
+        ch_types = ['eyetrack'] * n_chan  # ch_types = ['eyetrack'] * n_chan
+
+        info = create_info(ch_names, sfreq, ch_types)
+
+        # set correct channel type and location
+        loc = np.array([np.nan, np.nan, np.nan, np.nan, np.nan, np.nan,
+                        np.nan, np.nan, np.nan, np.nan, np.nan, np.nan])
+        for i_ch, ch_name in enumerate(ch_names):
+            coil_type = (FIFF.FIFFV_COIL_EYETRACK_PUPIL if ('Pupil' in ch_name)
+                         else FIFF.FIFFV_COIL_EYETRACK_POS)
+            unit = (FIFF.FIFF_UNIT_UNITLESS if ('Pupil' in ch_name)
+                    else FIFF.FIFF_UNIT_PX)
+
+            loc[3] = (-1 if ('L' in ch_name) else
+                      1 if ('R' in ch_name) else
+                      np.nan)
+            loc[4] = (-1 if ('X' in ch_name) else
+                      1 if ('Y' in ch_name) else
+                      np.nan)
+            info['chs'][i_ch]['coil_type'] = coil_type
+            info['chs'][i_ch]['loc'] = loc.copy()
+            info['chs'][i_ch]['unit'] = unit
+
+        # load data
+        df_recalibration, df_msg, df_fix, df_sacc, df_blink, df_samples = \
+            ParseEyeLinkAsc_(fname)
+
+        # clean out misread data
+        # assert tSample > 0
+        df_samples = df_samples[df_samples.tSample > 0]
+        # clean out rows with duplicate sampletime, as these are prob errors
+        # tbd
+        # also clean out rows where number of nans doesnt fit!
+        # nan_lines = df_samples.isna().any(axis=1)
+        # tmp = df_samples[nan_lines]
+
+        # mod_factor = 2 if (pos and not pupil) else 3 if (pos and pupil) else1
+        # (df_samples.isna().sum(axis=1) % mod_factor) != 0
+
+        # transpose to correct sfreq
+        # samples = df_samples['tSample'].apply(lambda x: x*sfreq/1000.)
+        # first_sample = samples.min()
+        shiftfun = lambda a, b: a + b - (a % b) if (a % b != 0) else a
```

Comment:
I'm not sure what this purpose of this lambda function was? Anyways I think it would only work as expected with data that have a 500Hz sampling frequency. With a 1000Hz frequency, every other sample would have it's timestamp shifted by 1 ms, and this would create duplicate timestamps.